dvc/output/base.py

```diff
@@ -669,18 +636,23 @@ def get_used_cache(self, **kwargs):
                     )
                 )
             logger.warning(msg)
-            return NamedCache()
+            return set()
 
-        ret = NamedCache.make(self.scheme, self.hash_info.value, str(self))
+        if self.is_dir_checksum:
+            return self.collect_used_dir_cache(**kwargs)
 
-        if not self.is_dir_checksum:
-            return ret
+        obj = self.get_obj(filter_info=kwargs.get("filter_info"))
+        if not obj:
+            obj = self.odb.get(self.hash_info)
 
-        ret.add_child_cache(
-            self.hash_info.value, self.collect_used_dir_cache(**kwargs),
-        )
+        return {obj}
 
-        return ret
+    def get_used_external(self, **kwargs) -> Dict["RepoPair", str]:
```

Comment:
seems like we probably want to split the idea of "used objects" and "used external deps" all the way up our call stack, but for now its only done in `output` (with the stage/repo level `used_cache` functions returning tuples of `used_objects, used_external_deps`)